tools/plugins/com.liferay.ide.upgrade.core/src/com/liferay/blade/eclipse/provider/ProjectMigrationService.java

```diff
@@ -392,6 +392,14 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
 					return FileVisitResult.SKIP_SUBTREE;
 				}
 
+				if (dir.endsWith("out") && dir.startsWith(startDir.getPath())) {
```

Comment:
This test will find things that are not wrong

Imagine if a user has a java package like this

`package com.liferay.my.project.build;`
or
`package com.liferay.my.project.bin;`

Both of those packages will have a directory called 'build' and 'bin' and in there will be `*.java` files.  If your change was made, then it would match those directories and they would be ignored.  That is bad.

So we have to make sure that the 'out/bin/build' are first level children of the project root directory

Reply:
cc @jtydhr88